telemetry/telemetry.py

```diff
@@ -0,0 +1,57 @@
+# Notebook Telemetry
+#
+# Elastic relies on anonymous usage statistics to continuously improve. This
+# script allows us to see which notebooks are most used, which in turn helps us
+# redirect our efforts to the areas that matter most to our users.
+#
+# What data is collected?
+#
+# This script configures your elasticsearch client to send a custom user agent
+# string with requests made to Elastic Cloud. The new user agent only contains
+# the name of the notebook. Here is an example user agent:
+#
+#     searchlabs/00-quick-start
+#
+# No other data is collected by this script.
+
+import os
+
+
+def get_notebook_name():
+    """Return the name of the running notebook."""
+    name = "unknown"
+
+    # first we try to get the name from the Jupyter environment
+    if os.environ.get("JPY_SESSION_NAME"):
+        name = os.path.basename(os.environ["JPY_SESSION_NAME"])
+
+    # next we check for Visual Studio Code metadata
+    elif "__vsc_ipynb_file__" in globals():
+        name = os.path.basename(globals()["__vsc_ipynb_file__"])
+    # else we try to get it from the Colab environment
+    else:
+        import requests
+
+        try:
+            name = requests.get("http://172.28.0.12:9000/api/sessions").json()[0][
+                "name"
+            ]
+        except Exception:
+            pass
+
+    return name.split(".ipynb")[0].lower()
+
+
+def enable_telemetry(client, notebook_name=None):
+    """Enable telemetry for the given elasticsearch client instance."""
+    if "nbtest" in os.environ.get("_", ""):
+        # no telemetry for tests
+        return
+
+    if notebook_name is None:
+        notebook_name = get_notebook_name()
+
+    client.options(
+        headers={**client._headers, "User-Agent": f"searchlabs/{notebook_name}"}
```

Comment:
Just a thought: I'm not sure if there is lots of value in this, but should we do best-effort extraction of the platform (with the code above) and add it to the agent, e.g.
`searchlabs/{notebook_name} jupyter/7.1.2`
`searchlabs/{notebook_name} vscode`
`searchlabs/{notebook_name} colab`

We would learn something about the environments of our users. In my current understanding, the KPI dashboards would keep working the same, but we would gain the possibility to ad-hoc check this (or create another dashboard one day).

Reply:
I think the correct thing to do here would be to do something like:

    User-Agent: searchlabs/00-quick-start (Jupyter 1.0.0)

Right? This is what browsers do, they stick a lot of platform specific stuff in the comments section of the agent string. I like this idea. I don't think we'll be able to add this extra info on charts though, so I'll ask some questions to see if there is any alternative format that would allow us to include the platform in our charts.

Reply:
one thing is to check with clients team on how they will treat the user-agent. I believe "search labs" will be product name and "00-quick-start" will be version name. 

I would like to aggregate on the version name in this example, to see the list of most popular notebooks that have been run. I wouldn't want the aggregation to be split when we include the platform.

Reply:
Yes. I have sent some test requests with the platform. Need to wait about 3 hours for them to show up in our telemetry and see how things look. I'll decide to include it or not depending on that.